core/unit_test/TestLocalDeepCopy.hpp

```diff
@@ -1042,4 +1042,74 @@ TEST(TEST_CATEGORY, local_deepcopy_rangepolicy_layoutright) {
   }
 }
 #endif
+
+namespace Impl {
+template <typename T, typename SHMEMTYPE>
+using ShMemView =
+    Kokkos::View<T, Kokkos::LayoutRight, SHMEMTYPE, Kokkos::MemoryUnmanaged>;
+
+struct DeepCopyScratchFunctor {
+  DeepCopyScratchFunctor(
+      Kokkos::View<double**, TEST_EXECSPACE::memory_space> check_view_1,
+      Kokkos::View<double**, TEST_EXECSPACE::memory_space> check_view_2)
+      : check_view_1_(check_view_1),
+        check_view_2_(check_view_2),
+        N_(check_view_1.extent(0)) {}
+
+  KOKKOS_INLINE_FUNCTION void operator()(
+      Kokkos::TeamPolicy<TEST_EXECSPACE,
+                         Kokkos::Schedule<Kokkos::Dynamic>>::member_type team)
+      const {
+    using ShmemType = TEST_EXECSPACE::scratch_memory_space;
+    Impl::ShMemView<double**, ShmemType> shview =
+        Kokkos::subview(Impl::ShMemView<double***, ShmemType>(
+                            team.team_scratch(1), team.team_size(), N_, N_),
+                        team.team_rank(), Kokkos::ALL(), Kokkos::ALL());
+
+    Kokkos::Experimental::local_deep_copy(team, shview, 5.0);
```

Comment:
This doesn't work either does it? The 5.0 would only be copied into a portion of each threads shview, since the loop to do so will be parallelized, but each thread works on different output.